### DISCUSSION TITLE
rust lsp didn't work

### DISCUSSION THREAD

**Discussion by Abdullah-Mahthel** [helix-editor/helix#11116] [General]
Hello guys ;

![rustimageissu](https://github.com/helix-editor/helix/assets/130181169/7d931c10-f406-4665-b152-4e00c2619a3c)

as you can see I am using windows ;

when I open rust file the server exited (you can see this message on the bottom left ) when opening rs file in helix

l don't know how to fix it

**Comment by kirawi** [comment]
Follow https://github.com/helix-editor/helix/wiki/Language-Server-Configurations#rust if you haven't already. If it still doesn't wotk, run `hx -v` and open a rust file. Then do `:log-open`

**Reply by Abdullah-Mahthel** [reply]
![rustdamnissue](https://github.com/helix-editor/helix/assets/130181169/0e988f2c-e02a-4bcc-9d37-7853c30eedcd)

l got 8k line of code when l open rust file then do :log-open :/

**Reply by kirawi** [reply]
You have to install `rust-analyzer` for your toolchain per line 88466. Each toolchain needs its own `rust-analyzer` component. Next time, please copy the text rather than creating an image since most of it is cut off.

**Reply by Abdullah-Mahthel** [reply]
rustup toolchain list
stable-x86_64-pc-windows-gnu (default)

this is my toolchain

**Reply by Abdullah-Mahthel** [reply]
![rs01](https://github.com/helix-editor/helix/assets/130181169/e215f48c-db53-488a-8ac8-dd295ddbd174)

this message

**Reply by kirawi** [reply]
I was referring to the log. Like I said, you have to reinstall it for your toolchain per https://github.com/helix-editor/helix/wiki/Language-Server-Configurations#rust

**Reply by Abdullah-Mahthel** [reply]
l don't know what l miss l can work with : C++ - C - Python - Even html -css but rust No

**Reply by kirawi** [reply]
What have you tried so far? Have you already tried what I suggested?

**Reply by Abdullah-Mahthel** [reply]
This is my languages.toml for rust : 

[[language]]
name = "rust"
[language-server.rust-analyzer.config.check]
command = "clippy"

scope = "source.rust"
injection-regex = "rs|rust"
file-types = ["rs"]
roots = ["Cargo.toml", "Cargo.lock"]
shebangs = ["rust-script", "cargo"]
auto-format = true
comment-tokens = ["//", "///", "//!"]

language-servers = [ "rust-analyzer" ]
indent = { tab-width = 4, unit = "    " }
persistent-diagnostic-sources = ["rustc", "clippy"]

[language.auto-pairs]
'(' = ')'
'{' = '}'
'[' = ']'
'"' = '"'
'`' = '`'

[language.debugger]
name = "lldb-dap"
transport = "stdio"
command = "lldb-dap"

[[language.debugger.templates]]
name = "binary"
request = "launch"
completion = [ { name = "binary", completion = "filename" } ]
args = { program = "{0}" }

[[language.debugger.templates]]
name = "binary (terminal)"
request = "launch"
completion = [ { name = "binary", completion = "filename" } ]
args = { program = "{0}", runInTerminal = true }

[[language.debugger.templates]]
name = "attach"
request = "attach"
completion = [ "pid" ]
args = { pid = "{0}" }

[[language.debugger.templates]]
name = "gdbserver attach"
request = "attach"
completion = [ { name = "lldb connect url", default = "connect://localhost:3333" }, { name = "file", completion = "filename" }, "pid" ]
args = { attachCommands = [ "platform select remote-gdb-server", "platform connect {0}", "file {1}", "attach {2}" ] }

[[grammar]]
name = "rust"
source = { git = "https://github.com/tree-sitter/tree-sitter-rust", rev = "9c84af007b0f144954adb26b3f336495cbb320a7" }

**Reply by Abdullah-Mahthel** [reply]
my toolchain : 
 rustup  toolchain list
stable-i686-pc-windows-msvc
stable-x86_64-pc-windows-msvc (default)

**Reply by Abdullah-Mahthel** [reply]
λ helix --health rust
Configured language servers:
  ✓ rust-analyzer: C:\Users\saqry\scoop\apps\rustup-gnu\current\.cargo\bin\rust-analyzer.exe
Configured debug adapter: lldb-dap
Binary for debug adapter: C:\Users\saqry\scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\lldb-dap.exe
Configured formatter: None
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓

**Reply by kirawi** [reply]
Run `rustup component list`

**Reply by Abdullah-Mahthel** [reply]
λ rustup component list                        
cargo-x86_64-pc-windows-msvc (installed)       
clippy-x86_64-pc-windows-msvc (installed)      
llvm-bitcode-linker-x86_64-pc-windows-msvc     
llvm-tools-x86_64-pc-windows-msvc              
rls-x86_64-pc-windows-msvc                     
rust-analysis-x86_64-pc-windows-msvc           
rust-analyzer-x86_64-pc-windows-msvc           
rust-docs-x86_64-pc-windows-msvc (installed)   
rust-src                                       
rust-std-aarch64-apple-darwin                  
rust-std-aarch64-apple-ios                     
rust-std-aarch64-apple-ios-sim                 
rust-std-aarch64-linux-android                 
rust-std-aarch64-pc-windows-gnullvm            
rust-std-aarch64-pc-windows-msvc               
rust-std-aarch64-unknown-fuchsia               
rust-std-aarch64-unknown-linux-gnu             
rust-std-aarch64-unknown-linux-musl            
rust-std-aarch64-unknown-linux-ohos            
rust-std-aarch64-unknown-none                  
rust-std-aarch64-unknown-none-softfloat        
rust-std-aarch64-unknown-uefi                  
rust-std-arm-linux-androideabi                 
rust-std-arm-unknown-linux-gnueabi             
rust-std-arm-unknown-linux-gnueabihf           
rust-std-arm-unknown-linux-musleabi            
rust-std-arm-unknown-linux-musleabihf          
rust-std-armebv7r-none-eabi                    
rust-std-armebv7r-none-eabihf                  
rust-std-armv5te-unknown-linux-gnueabi         
rust-std-armv5te-unknown-linux-musleabi        
rust-std-armv7-linux-androideabi               
rust-std-armv7-unknown-linux-gnueabi           
rust-std-armv7-unknown-linux-gnueabihf         
rust-std-armv7-unknown-linux-musleabi          
rust-std-armv7-unknown-linux-musleabihf        
rust-std-armv7-unknown-linux-ohos              
rust-std-armv7a-none-eabi                      
rust-std-armv7r-none-eabi                      
rust-std-armv7r-none-eabihf                    
rust-std-i586-pc-windows-msvc                  
rust-std-i586-unknown-linux-gnu                
rust-std-i586-unknown-linux-musl               
rust-std-i686-linux-android                    
rust-std-i686-pc-windows-gnu                   
rust-std-i686-pc-windows-gnullvm               
rust-std-i686-pc-windows-msvc                  
rust-std-i686-unknown-freebsd                  
rust-std-i686-unknown-linux-gnu                
rust-std-i686-unknown-linux-musl               
rust-std-i686-unknown-uefi                     
rust-std-loongarch64-unknown-linux-gnu         
rust-std-loongarch64-unknown-none              
rust-std-loongarch64-unknown-none-softfloat    
rust-std-nvptx64-nvidia-cuda                   
rust-std-powerpc-unknown-linux-gnu             
rust-std-powerpc64-unknown-linux-gnu           
rust-std-powerpc64le-unknown-linux-gnu         
rust-std-riscv32i-unknown-none-elf             
rust-std-riscv32im-unknown-none-elf            
rust-std-riscv32imac-unknown-none-elf          
rust-std-riscv32imafc-unknown-none-elf         
rust-std-riscv32imc-unknown-none-elf           
rust-std-riscv64gc-unknown-linux-gnu           
rust-std-riscv64gc-unknown-none-elf            
rust-std-riscv64imac-unknown-none-elf          
rust-std-s390x-unknown-linux-gnu               
rust-std-sparc64-unknown-linux-gnu             
rust-std-sparcv9-sun-solaris                   
rust-std-thumbv6m-none-eabi                    
rust-std-thumbv7em-none-eabi                   
rust-std-thumbv7em-none-eabihf                 
rust-std-thumbv7m-none-eabi                    
rust-std-thumbv7neon-linux-androideabi         
rust-std-thumbv7neon-unknown-linux-gnueabihf   
rust-std-thumbv8m.base-none-eabi               
rust-std-thumbv8m.main-none-eabi               
rust-std-thumbv8m.main-none-eabihf             
rust-std-wasm32-unknown-emscripten             
rust-std-wasm32-unknown-unknown                
rust-std-wasm32-wasi                           
rust-std-wasm32-wasip1                         
rust-std-wasm32-wasip1-threads                 
rust-std-x86_64-apple-darwin                   
rust-std-x86_64-apple-ios                      
rust-std-x86_64-fortanix-unknown-sgx           
rust-std-x86_64-linux-android                  
rust-std-x86_64-pc-solaris                     
rust-std-x86_64-pc-windows-gnu                 
rust-std-x86_64-pc-windows-gnullvm             
rust-std-x86_64-pc-windows-msvc (installed)    
rust-std-x86_64-unknown-freebsd                
rust-std-x86_64-unknown-fuchsia                
rust-std-x86_64-unknown-illumos                
rust-std-x86_64-unknown-linux-gnu              
rust-std-x86_64-unknown-linux-gnux32           
rust-std-x86_64-unknown-linux-musl             
rust-std-x86_64-unknown-linux-ohos             
rust-std-x86_64-unknown-netbsd                 
rust-std-x86_64-unknown-none                   
rust-std-x86_64-unknown-redox                  
rust-std-x86_64-unknown-uefi                   
rustc-x86_64-pc-windows-msvc (installed)       
rustc-dev-aarch64-apple-darwin                 
rustc-dev-aarch64-pc-windows-msvc              
rustc-dev-aarch64-unknown-linux-gnu            
rustc-dev-aarch64-unknown-linux-musl           
rustc-dev-arm-unknown-linux-gnueabi            
rustc-dev-arm-unknown-linux-gnueabihf          
rustc-dev-armv7-unknown-linux-gnueabihf        
rustc-dev-i686-pc-windows-gnu                  
rustc-dev-i686-pc-windows-msvc                 
rustc-dev-i686-unknown-linux-gnu               
rustc-dev-loongarch64-unknown-linux-gnu        
rustc-dev-powerpc-unknown-linux-gnu            
rustc-dev-powerpc64-unknown-linux-gnu          
rustc-dev-powerpc64le-unknown-linux-gnu        
rustc-dev-riscv64gc-unknown-linux-gnu          
rustc-dev-s390x-unknown-linux-gnu              
rustc-dev-x86_64-apple-darwin                  
rustc-dev-x86_64-pc-windows-gnu                
rustc-dev-x86_64-pc-windows-msvc               
rustc-dev-x86_64-unknown-freebsd               
rustc-dev-x86_64-unknown-illumos               
rustc-dev-x86_64-unknown-linux-gnu             
rustc-dev-x86_64-unknown-linux-musl            
rustc-dev-x86_64-unknown-netbsd                
rustc-docs-x86_64-unknown-linux-gnu            
rustfmt-x86_64-pc-windows-msvc (installed)

**Reply by kirawi** [reply]
`rust-analyzer-x86_64-pc-windows-msvc`
It's not installed for your toolchain. Like I've been saying, run `rustup component add rust-analyzer`

**Reply by Abdullah-Mahthel** [reply]
it worked :)  the server work ok 

auto completion works but no colors :/

**Reply by Abdullah-Mahthel** [reply]
Do you know How to fix colors and How can l copy from web and paste to helix?
